Library/Tests/ViewModels/SearchViewModelTests.swift

```diff
@@ -455,4 +464,51 @@ internal final class SearchViewModelTests: TestCase {
         self.trackingClient.events)
     }
   }
+
+  func testScrollAndUpdateProjects_ViaProjectNavigator() {
+    let playlist = (0...10).map { idx in .template |> Project.lens.id .~ (idx + 42) }
+    let projectEnv = .template
+      |> DiscoveryEnvelope.lens.projects .~ playlist
+
+    let playlist2 = (0...20).map { idx in .template |> Project.lens.id .~ (idx + 82) }
+    let projectEnv2 = .template
+      |> DiscoveryEnvelope.lens.projects .~ playlist2
+
+    withEnvironment(apiService: MockService(fetchDiscoveryResponse: projectEnv)) {
+      self.vm.inputs.viewWillAppear(animated: false)
+
+      self.scheduler.advance()
+
+      self.hasAddedProjects.assertValues([true], "Projects are loaded.")
+
+      self.vm.inputs.tapped(project: playlist[4])
+
+      self.vm.inputs.transitionedToProject(at: 5, outOf: playlist.count)
+
+      self.scrollToProjectRow.assertValues([5])
+
+      self.vm.inputs.transitionedToProject(at: 6, outOf: playlist.count)
+
+      self.scrollToProjectRow.assertValues([5, 6])
+
+      self.vm.inputs.transitionedToProject(at: 7, outOf: playlist.count)
+
+      self.scrollToProjectRow.assertValues([5, 6, 7])
+
+      withEnvironment(apiService: MockService(fetchDiscoveryResponse: projectEnv2)) {
+        self.vm.inputs.transitionedToProject(at: 8, outOf: playlist.count)
+
+        self.scheduler.advance()
+
+        self.scrollToProjectRow.assertValues([5, 6, 7, 8])
+
+        // this works in practice, gotta figure out what i'm missing in the test.
+        self.hasAddedProjects.assertValues([true, true], "More projects are loaded.")
+
```

Comment:
Same test in DiscoveryPage and Profile, but it fails here (works in the sim/device though). Just need to step away for a hot min and think about what's missing.